atomic_reactor/plugins/post_flatpak_create_oci.py

```diff
@@ -68,119 +34,63 @@ def __init__(self, workflow):
         :param workflow: DockerBuildWorkflow instance
         """
         super(FlatpakCreateOciPlugin, self).__init__(workflow)
-        self.builder = None
 
         try:
             self.flatpak_metadata = self.workflow.conf.flatpak_metadata
         except KeyError:
             self.flatpak_metadata = FLATPAK_METADATA_ANNOTATIONS
 
-#    def _export_container(self, container_id):
-#        export_generator = self.tasker.export_container(container_id)
-#        export_stream = StreamAdapter(export_generator)
-#
-#        outfile, manifestfile = self.builder._export_from_stream(export_stream)
-#
-#        return outfile, manifestfile
-
-    def _export_filesystem(self):
-        # image = self.workflow.image
-        self.log.info("Creating temporary docker container")
-        # The command here isn't used, since we only use the container for export,
-        # but (in some circumstances) the docker daemon will error out if no
-        # command is specified.
-
-        # no longer create and export container, just use oc image extract
-        # to get content of the image
-        # OSBS2 TBD
-        # container_dict = self.tasker.create_container(image, command=["/bin/bash"])
-        # container_id = container_dict['Id']
-
-        # try:
-        #     return self._export_container(container_id)
-        # finally:
-        #     self.log.info("Cleaning up docker container")
-        #     self.tasker.remove_container(container_id)
-        return None, None
-
-    def _get_oci_image_id(self, oci_path):
-        cmd = ['skopeo', 'inspect', '--raw', 'oci:{}'.format(oci_path)]
-        raw_manifest = retries.run_cmd(cmd)
-        oci_image_manifest = json.loads(raw_manifest)
-        return oci_image_manifest['config']['digest']
-
-    def _copy_oci_to_local_storage(self, oci_path, name, tag):
-        """Copy OCI image to internal container storage
-
-        The internal storage to copy the image to is defined by the workflow.
-
-        :param oci_path: str, path to OCI directory
-        :param name: str, name to be given to the image in the internal storage
-        :param tag: str, tag to apply to name in the internal storage
-        """
-        skopeo_copy_dst = '{}:{}:{}'.format(self.workflow.storage_transport, name, tag)
-        cmd = ['skopeo', 'copy', 'oci:{}'.format(oci_path), skopeo_copy_dst]
+    def build_flatpak_image(
+            self, source, build_dir: BuildDir
+    ) -> Dict[str, Union[Dict[str, Union[str, int]], List[Dict[str, Union[str, int]]]]]:
```

Comment:
When it gets this complicated, I feel like a `Dict[str, Any]` annotation would be better

The perfect solution would be a [TypedDict](https://docs.python.org/3/library/typing.html#typing.TypedDict), but we haven't used them anywhere else yet

Reply:
updated